### PR TITLE
fix(components/core): hide viewkeeper overflow

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2036346](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2036346)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/880/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/ag-grid-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/layout-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/tabs-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/880/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/880/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/880/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/880/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/880/playground/)

**Comment by johnhwhite** [comment]
Examples to check:

* [modals](https://blackbaud.github.io/skyux-pr-preview/880/playground/#/components/modal)
* [flyout](https://blackbaud.github.io/skyux-pr-preview/880/playground/#/components/flyout/flyout)
* [split view](https://blackbaud.github.io/skyux-pr-preview/880/playground/#/components/split-view/split-view/page-bound)

Scroll to where the viewkeeper element starts to go out of view. That element now has a `clip-path` because `position: fixed` elements will go past `overflow` bounds.

image

File: libs/components/core/src/lib/modules/viewkeeper/viewkeeper.ts [hunk 1]
```diff
@@ -67,15 +67,17 @@ function setElPosition(
...
   el.style.top = px(top);
   el.style.left = px(left);
   el.style.marginTop = px(marginTop);
+  el.style.clipPath =
+    clipTop || clipLeft ? `inset(${px(clipTop)} 0 0 ${px(clipLeft)})` : 'none';
 
-  /*istanbul ignore else*/
-  /* sanity check */
-  if (width !== null) {
+  if (width) {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/viewkeeper/viewkeeper.ts]: Before, this would allow a zero-width value. We don't want that anymore?

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
I'm unsure if this is related to this pull request, but I noticed that the border on the toolbar shrinks from 2 pixels to 1 pixel when it is "viewkept".
Screen Shot 2022-12-12 at 10 38 18 AM
Screen Shot 2022-12-12 at 10 38 23 AM

File: libs/components/core/src/lib/modules/viewkeeper/viewkeeper.ts (hunk 1 above)
Comment (johnhwhite) [reply libs/components/core/src/lib/modules/viewkeeper/viewkeeper.ts]: Reverted this change.

**Comment by johnhwhite** [comment]
> I'm unsure if this is related to this pull request, but I noticed that the border on the toolbar shrinks from 2 pixels to 1 pixel when it is "viewkept". Screen Shot 2022-12-12 at 10 38 18 AM Screen Shot 2022-12-12 at 10 38 23 AM

The border comes from elements that are not "viewkept" — there is a viewkeeper on the toolbar above the grid, there is a viewkeeper on the header row within the grid, and when they are both viewkept the header row is kept below the height of the toolbar.

To avoid the shift, the style would need to be adjusted, possibly moving the border to the header row rather than on the whole grid. I'm inclined to address that issue separately as it may get complicated.

**Comment by johnhwhite** [comment]
Filed a bug [AB#2391893](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2391893) to follow up.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**